internal/gengapic/genrest.go

```diff
@@ -838,12 +838,19 @@ func (g *generator) unaryRESTCall(servName string, m *descriptor.MethodDescripto
 	p("  return nil, err")
 	p("}")
 	p("")
-	p("unm := protojson.UnmarshalOptions{AllowPartial: true, DiscardUnknown: true}")
-	p("rsp := &%s.%s{}", outSpec.Name, outType.GetName())
-	p("")
-	p("if err := unm.Unmarshal(buf, rsp); err != nil {")
-	p("  return nil, maybeUnknownEnum(err)")
-	p("}")
+	if outSpec.Name == "httpbodypb" && outType.GetName() == "HttpBody" {
+		p("rsp := &httpbodypb.HttpBody{")
+		p("  Data: buf,")
+		p("  ContentType: httpRsp.Header[\"Content-Type\"][0],")
+		p("}")
```

Comment:
```suggestion
		p("}")
		p(`if headers := httpRsp.Header; len(headers["Content-Type"]) > 0 {`)
		p(`  rsp.ContentType = headers["Content-Type"][0]`)
		p("}")
```

Notice the use of back ticks instead of double-quotes so that we can use double-quotes unescaped in the string.

Reply:
Considering there could be more than one value for `Content-Type`, should we join on `,` and return "all values" of the header? IDK if comma-separator is how HTTP delineates multiple values, or if its `;`, but we should do that if at all.

Reply:
I'm not sure that multiple Content-Type headers are actually possible - it might just be an artifact of the implementation. https://stackoverflow.com/questions/5809099/does-the-http-protocol-support-multiple-content-types-in-response-headers

Reply:
Cool thanks for looking into it.